docs/api-reference/index.md

```diff
@@ -0,0 +1,35 @@
+# API Reference
+
+## Classes
+
+```{eval-rst}
+.. currentmodule:: esssans
+
+.. autosummary::
+   :toctree: ../generated/classes
+   :template: class-template.rst
+   :recursive:
+```
+
+## Top-level functions and attributes
+
+```{eval-rst}
+.. autosummary::
+   :toctree: ../generated/functions
+   :recursive:
+
+   providers
+```
+
+## Submodules
+
+```{eval-rst}
+.. autosummary::
+   :toctree: ../generated/modules
+   :template: module-template.rst
+   :recursive:
+
+   beam_center_finder
+   sans2d
+   types
```

Comment:
Was it intentional that some entries under "Module attributes" and others under "Classes"? I don't really seem to get the logic separating the two sections.

Reply:
It is not intentional, but this is what we get from the generic template, and due to the fact that some domain types are using `NewType` while others are not (are actual classes). I don't know how to fix that. Maybe the template could be rewritten?